notebooks/query-language-object-based-dialect/01.01.quantifier_capabilities.ipynb

```diff
@@ -0,0 +1,842 @@
+   "id": "3bc1a4c7",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "gf.dataframe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f8f3341c-afc8-43c9-915a-ec6bc250cbf3",
+   "metadata": {},
+   "source": [
+    "### Dropping index levels\n",
+    "\n",
+    "As a precursor to defining queries, we drop the index level of the GraphFrame using the `drop_index_levels()` Hatchet function. Hatchet hierarchical indexing can be of two types, depending on whether there is a single metric per node or multiple set of metrics per node.  \n",
+    "\n",
+    "If a node contains a single metric, the DataFrame will use an `Index` object containing the node column. If a node has additional level of information, Hatchet creates a `MultiIndex` to store the information pertaining to multiple set of metrics per node. `MultiIndex` stores the node column as the \"top\" level of the index, followed by additional information on the levels below. \n",
+    "\n",
+    "Based on the types of indexing (`Index or MultiIndex`), retrieving data from a DataFrame corresponding to a particular node either retrieves a single or multiple rows. This difference can cause issues when applying query node predicates.\n",
+    "Therefore, it is necessary to get rid of all index levels besides the node column through an aggregation operation on the GraphFrame. Then, a query node predicate can be applied to the GraphFrame. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f216dbef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gf.drop_index_levels()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fec2520f",
+   "metadata": {},
+   "source": [
+    "### Query type 1: Match a single node\n",
+    "\n",
+    "In the simpliest case, one can use this query type to match all single nodes that belong to function calls of a particular library (e.g., MPI). "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "373ad780",
+   "metadata": {},
+   "source": [
+    "The following query matches all single nodes where the predicate that the `name` metric `starts with \"MPI_Barrier\"` is satisfied:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "455adb46",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# single node with names starting with MPI_Barrier\n",
+    "query_1 = [\n",
+    "    (1, {\"name\": \"MPI_Barrier.*\"}),\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "82a5a9c6",
+   "metadata": {},
+   "source": [
+    "The above query, when used with an accepted value for the quantifier, \".\", to match all single nodes with a certain predicate, is equivalent to the follwing query:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a1adf78c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# single node with names starting with MPI_Barrier\n",
+    "query_1 = [\n",
```

Comment:
I would probably separate the queries, as `query_1_1 = [(1, {"name": "MPI_Wait.*"})]` and `query_1_2=[(".", {"name": "MPI_Wait.*"})]`. Also I couldnt fully understand the query_1_1. What difference does providing a number make?